go/autoscaler/autoscaler.go

```diff
@@ -505,7 +498,9 @@ func (a *Autoscaler) Monitor() {
 					continue
 				}
 				j.TrainerJob = tj
+				a.jobs[key] = j
```

Comment:
Need to write back to the map, `j.TrainerJob = tj` is not sufficient.

Reply:
Refine the code later is fine. Also, the scaling algorithm should be updated like: 

1. `ScaleDryRun` search in order to find a sufficient node to scale up 1 pod, return the diff.
1. Cluster total resource is *not* used for the scale-up, use only per node resource.
1. Find a way to ensure `max_load_desired` for each node.

Reply:
Thanks!

Btw, maybe `max_load_desired` is for the entire cluster? Otherwise if it's for each node on max_load_desired=0.97, each node will have 0.03*12=0.36 CPUs, probably many pod can not be scheduled on it anyway.